### DISCUSSION TITLE
Ability to strongly type parameter properties

### DISCUSSION THREAD

**Discussion by jdmcd** [remix-run/react-router#12614] [Proposals]
It would be cool to be able to do something like this:

```ts
route('edit/int(:id)', './routes/admin/edit-action.tsx'),
```

And then the generated `params` object would be:

```ts
{ id: number }
```

**Comment by sergiodxa** [comment]
All route params are always strings

**Reply by jdmcd** [reply]
I know, but I'm suggesting as a proposal that it would be nice to be able to strongly type the properties based on user definition.

**Reply by gustavopch** [reply]
They'll do it AFAIK.
